includes/class-wc-stripe-helper.php

```diff
@@ -726,12 +726,21 @@ public static function filter_payment_methods_with_capabilities( $payment_method
 			return [];
 		}
 
+		// Return all payment methods if in test mode.
+		if ( $testmode ) {
+			return $payment_method_ids;
+		}
+
 		$payment_method_ids_with_capability = [];
 
 		foreach ( $payment_method_ids as $payment_method_id ) {
 			$key            = $payment_method_id . '_payments';
-			$has_capability = isset( $data['capabilities'][ $key ] );
-			if ( $has_capability || $testmode ) {
+			// Check if the payment method has capabilities set in the account data.
+			// Generally the key is the payment method id appended with '_payments' (i.e. 'card_payments', 'sepa_debit_payments', 'klarna_payments').
+			// In some cases, the Stripe account might have the legacy key set. For example, for Klarna, the legacy key is 'klarna'.
+			// For card, the legacy key is 'legacy_payments'.
+			$has_capability = isset( $data['capabilities'][ $key ] ) || isset( $data['capabilities'][ $payment_method_id ] ) || ( 'card' === $payment_method_id && $data['capabilities']['legacy_payments'] );
```

Comment:
```suggestion
			$has_capability = isset( $data['capabilities'][ $key ] ) || isset( $data['capabilities'][ $payment_method_id ] ) || ( 'card' === $payment_method_id && isset( $data['capabilities']['legacy_payments'] ) );
```

There's a missing isset check on the final `legacy_payments`, if it ever got to that condition it would error if it wasn't set. 

```
PHP Warning:  Undefined array key "legacy_payments" in /wp-content/plugins/woocommerce-gateway-stripe/includes/class-wc-stripe-helper.php on line 746

Reply:
That's a good catch @james-allan 😬 
Fixed in [132cbca](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3375/commits/132cbcafeec9379a8267f734e5c49aa716937d43)